### PR TITLE
Support `MessageBodyWriter` and partially `MessageBodyReader`

### PR DESCRIPTION
**Pull request by dstepanov**
Also added some dummy request objects for now to prevent errors

### CODE REVIEW

File: jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsContextHttpHeaders.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+/**
+ * The bean context implementation of {@link jakarta.ws.rs.core.HttpHeaders}.
+ *
+ * @author Denis Stepanov
+ * @since 4.6.0
+ */
+@Internal
+@Singleton
+public final class JaxRsContextHttpHeaders implements jakarta.ws.rs.core.HttpHeaders {
```
Comment (graemerocher) [inline jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsContextHttpHeaders.java]: why is this class needed and does it need to be public?

File: jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsContextRequest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * The bean context implementation of {@link Request}.
+ *
+ * @author Denis Stepanov
+ * @since 4.6.0
+ */
+@Internal
+@Singleton
+public final class JaxRsContextRequest implements Request {
```
Comment (graemerocher) [inline jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsContextRequest.java]: why is this class needed and does it need to be public?

File: jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsContextUriInfo.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+/**
+ * The bean context implementation of {@link UriInfo}.
+ *
+ * @author Denis Stepanov
+ * @since 4.6.0
+ */
+@Internal
+@Singleton
+public final class JaxRsContextUriInfo implements UriInfo {
```
Comment (graemerocher) [inline jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsContextUriInfo.java]: why is this class needed and does it need to be public?

File: jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsHeadersMultivaluedMap.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+/**
+ * The representation of headers as {@link MultivaluedMap}.
+ *
+ * @author Denis Stepanov
+ * @since 4.6.0
+ */
+@Internal
+@Singleton
+public class JaxRsHeadersMultivaluedMap extends AbstractMap<String, List<String>> implements MultivaluedMap<String, String> {
```
Comment (graemerocher) [inline jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsHeadersMultivaluedMap.java]: why does this need to be a bean and public?

**Review by graemerocher — APPROVED** [review]

File: jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsContextHttpHeaders.java (hunk 1 above)
Comment (dstepanov) [reply jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsContextHttpHeaders.java]: Removed

File: jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsContextRequest.java (hunk 2 above)
Comment (dstepanov) [reply jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsContextRequest.java]: Removed

File: jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsContextUriInfo.java (hunk 3 above)
Comment (dstepanov) [reply jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsContextUriInfo.java]: Removed

File: jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsHeadersMultivaluedMap.java (hunk 4 above)
Comment (dstepanov) [reply jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsHeadersMultivaluedMap.java]: Removd

**Pull request merged**